src/main/java/org/openstreetmap/atlas/geography/atlas/sqlite/SQLiteItemWriter.java

```diff
@@ -0,0 +1,216 @@
+
+    SQLiteItemWriter(final Connection connection, final boolean buildIndices,
+            final PackedTagStore tagStore, final LongToLongMap identifierToTagStoreIndex,
+            final ItemType itemType) throws SQLException
+    {
+        this.tagStore = tagStore;
+        this.identifierToTagStoreIndex = identifierToTagStoreIndex;
+        this.itemType = itemType;
+        this.connection = connection;
+
+        // Construct the prepared statements
+        this.itemStatement = connection.prepareStatement(
+                String.format(INSERT_SQL, this.itemType.toString().toLowerCase()));
+        this.tagsStatement = connection.prepareStatement(
+                String.format(INSERT_TAGS_SQL, this.itemType.toString().toLowerCase()));
+        if (buildIndices)
+        {
+            this.rtreeStatementOptional = Optional.of(connection.prepareStatement(
+                    String.format(INSERT_RTREE_SQL, this.itemType.toString().toLowerCase())));
+        }
+        else
+        {
+            this.rtreeStatementOptional = Optional.empty();
+        }
+    }
+
+    public void write(final PackedAtlas atlas) throws SQLException
+    {
+        // This is very important. When you're doing bulk inserts, you need to turn this off.
+        // Performance is very bad without doing this!
+        this.connection.setAutoCommit(false);
+
+        final Iterable<? extends AtlasItem> items = getItemIterableOfCorrectType(atlas,
+                this.itemType);
+
+        for (final AtlasItem item : items)
+        {
+            // INSERT feature data
+            final long atlasId = item.getIdentifier();
+            this.itemStatement.setLong(FIRST_COLUMN, atlasId);
+            this.itemStatement.setString(SECOND_COLUMN, "md5sum_here");
+
+            final Optional<String> shardName = atlas.metaData().getShardName();
+            if (shardName.isPresent())
+            {
+                this.itemStatement.setString(THIRD_COLUMN, shardName.get());
+            }
+            else
+            {
+                this.itemStatement.setString(THIRD_COLUMN, null);
+            }
+
+            final byte[] wkb = getWKBOfItem(item);
+            this.itemStatement.setBytes(FOURTH_COLUMN, wkb);
+            this.itemStatement.execute();
+
+            // INSERT tag data for the feature
+            final long tagStoreIndex = this.identifierToTagStoreIndex.get(new Long(atlasId));
+            final int[] keyIndices = this.tagStore.getKeyArrayAtIndex(tagStoreIndex);
+            final int[] valueIndices = this.tagStore.getValueArrayAtIndex(tagStoreIndex);
+
+            if (keyIndices.length != valueIndices.length)
+            {
+                throw new CoreException(
+                        "key and value array length mismatch, something went wrong");
+            }
+
+            for (int index = 0; index < keyIndices.length; index++)
+            {
+                this.tagsStatement.setLong(FIRST_COLUMN, atlasId);
```

Comment:
This is one of the cases where I would argue that Magic Numbers should be ignored in the check style.